go/vt/sqlparser/predicate_rewriting_test.go

```diff
@@ -129,6 +129,17 @@ func TestRewritePredicate(in *testing.T) {
 	}, {
 		in:       "A and (B or A)",
 		expected: "A",
+	}, {
+		in: "(a = 1 and b = 41) or (a = 2 and b = 42)",
+		// this might look weird, but it allows the planner to either a or b in a vindex operation
+		expected: "a in (1, 2) and (a = 1 or b = 42) and ((b = 41 or a = 2) and b in (41, 42))",
+	}, {
+		in:       "(a = 1 and b = 41) or (a = 2 and b = 42) or (a = 3 and b = 43)",
+		expected: "a in (1, 2, 3) and (a in (1, 2) or b = 43) and ((a = 1 or b = 42 or a = 3) and (a = 1 or b = 42 or b = 43)) and ((b = 41 or a = 2 or a = 3) and (b = 41 or a = 2 or b = 43) and ((b in (41, 42) or a = 3) and b in (41, 42, 43)))",
+	}, {
+		// this has too many OR expressions in it, so we don't even try the CNF rewriting
+		in:       "a = 1 and b = 41 or a = 2 and b = 42 or a = 3 and b = 43 or a = 4 and b = 44 or a = 5 and b = 45",
+		expected: "a = 1 and b = 41 or a = 2 and b = 42 or a = 3 and b = 43 or a = 4 and b = 44 or a = 5 and b = 45",
```

Comment:
This test case has only 4 `or` expressions and is thus rewritten. I'll push a follow-up commit to fix this test + the failure in plan test.